### ISSUE TITLE
Sorting schema should not be generated for cypher fields with NonNullable arguments

### ISSUE THREAD

**Issue by mjfwebb** [neo4j/graphql#5631]
Given the following type definitions, schema is generated for sorting on the field `custom_string_with_param`. 

This should not occur as that field takes a NonNullable argument.

```graphql
type Movie {
    custom_string_with_param(param: String!): String!
        @cypher(statement: "RETURN $param as c", columnName: "c")
}
```

Excerpt from the schema generated:

```graphql
            \\"\\"\\"
            Fields to sort Movies by. The order in which sorts are applied is not guaranteed when specifying many fields in one MovieSort object.
            \\"\\"\\"
            input MovieSort {
              custom_string_with_param: SortDirection
            }
```

We should ensure that custom cypher fields with NonNullable arguments do not have sort schema generated.

**Comment by neo4j-team-graphql** [comment]
We've been able to confirm this bug using the steps to reproduce that you provided - many thanks @mjfwebb! :pray: We will now prioritise the bug and address it appropriately.

**Comment by MacondoExpress** [comment]
This will be fixed by [#5703](https://github.com/neo4j/graphql/pull/5703), 
note that there were other cases where `@cypher` fields were included in the schema, for instance, when a Node was returned, that is currently not sortable.

https://github.com/neo4j/graphql/pull/5703/files#diff-9c6f3a4b6930a1e9f5066937094c940f9a595caf2f1de3ecd21f9ff19d13b77eL330
